src/java/com/pb/mtctm2/abm/ctramp/ExplicitTelecommuteModel.java

```diff
@@ -0,0 +1,146 @@
+    public static final int        			   ET_MODEL_YES_ALT = 2;
+
+    
+    private ChoiceModelApplication etModel;
+    private ChoiceModelApplication updateCdapModel;
+    private ExplicitTelecommuteDMU etDmuObject;
+    
+
+    
+    public ExplicitTelecommuteModel( HashMap<String, String> propertyMap, CtrampDmuFactoryIf dmuFactory )
+    {               
+        setupExplicitTelecommuteChoiceModel(propertyMap, dmuFactory);
+    }
+
+    private void setupExplicitTelecommuteChoiceModel(HashMap<String, String> propertyMap, CtrampDmuFactoryIf dmuFactory)
+    {
+        logger.info("setting up explicit telecommute model.");
+
+        // locate the explicit telecommute  UEC
+        String uecFileDirectory = propertyMap.get( CtrampApplication.PROPERTIES_UEC_PATH );
+        String etUecFile = uecFileDirectory + propertyMap.get(ET_CONTROL_FILE_TARGET);
+
+        int dataSheet = Util.getIntegerValueFromPropertyMap( propertyMap, ET_DATA_SHEET_TARGET );
+        int modelSheet = Util.getIntegerValueFromPropertyMap( propertyMap, ET_MODEL_SHEET_TARGET );
+        int updateCdapModelSheet = Util.getIntegerValueFromPropertyMap(propertyMap, ET_UPDATE_CDAP_MODEL_SHEET_TARGET);
+        // create the explicit telecommute choice model DMU object.
+        etDmuObject = dmuFactory.getExplicitTelecoummteDMU();
+
+        // create the explicit telecommute choice model object
+        etModel = new ChoiceModelApplication(etUecFile, modelSheet, dataSheet, propertyMap, (VariableTable) etDmuObject);
+        updateCdapModel = new ChoiceModelApplication(etUecFile, updateCdapModelSheet, dataSheet, propertyMap, (VariableTable) etDmuObject);
+        
+        
+    }
+
+    
+    public void applyModel(Household hhObject){
+
+        Random hhRandom = hhObject.getHhRandom();
+        //Get pre-telecommute hh cdap and store it in a different object
+        hhObject.setPreEtCdapPattern(hhObject.getCoordinatedDailyActivityPattern());
+        // person array is 1-based
+        Person[] person = hhObject.getPersons();         
+        for (int i=1; i<person.length; i++) 
+        {
+            int workLoc = person[i].getUsualWorkLocation();
+            String cdap = person[i].getCdapActivity();
+            person[i].setPreExplicitTelecommuteCdap(cdap); //Storing the person pre-et cdap here 
+            int cdapIndex = person[i].getCdapIndex();
+            if ( workLoc != ModelStructure.WORKS_AT_HOME_LOCATION_INDICATOR && workLoc >0 && cdapIndex == 1 ) 
+            {
+                double randomNumber = hhRandom.nextDouble();
+                int chosen = getEtChoice(person[i], randomNumber);
+                person[i].setEtChoice(chosen);
+                if (chosen == 2) 
+                {
+                	String hhCdap = hhObject.getCoordinatedDailyActivityPattern();
+                	int person_index = person[i].getPersonNum();
+                	String newChosenCdapActivity = "N"; //if the HH pattern is joint, update cdap to N
```

Comment:
@DavidOry  you probably already thought about this - for 3+ households, say the initial cdap pattern is "MNNj", and the 1st person is telecommuting, technically both "NNNj" and "HNNj" are valid options. It's probably not a big deal as there are not that many large households with joint patterns and with telecommuters. It just means in the downstream we could have too many non-mandatory tours for telecommuters, which can be calibrated.

Reply:
That's a good point. Let's stick with this and see how the model run goes. As you say, it should be a very easy calibration adjustment to CDAP to get the total `N` outcomes right. @vivverma9: to the issue, can you please add the number of these cases that we have? Meaning, a telecommuter in a 3 plus person household in which another joint tour is possible (i.e., at least two other household members engaging in out of home travel).

Reply:
Here is a waterfall of such cases:
<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/USVV72~1/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/USVV72~1/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>
<!--table
	{mso-displayed-decimal-separator:"\.";
	mso-displayed-thousand-separator:"\,";}
@page
	{margin:.75in .7in .75in .7in;
	mso-header-margin:.3in;
	mso-footer-margin:.3in;}
tr
	{mso-height-source:auto;}
col
	{mso-width-source:auto;}
br
	{mso-data-placement:same-cell;}
td
	{padding-top:1px;
	padding-right:1px;
	padding-left:1px;
	mso-ignore:padding;
	color:black;
	font-size:11.0pt;
	font-weight:400;
	font-style:normal;
	text-decoration:none;
	font-family:Calibri, sans-serif;
	mso-font-charset:0;
	mso-number-format:General;
	text-align:general;
	vertical-align:bottom;
	border:none;
	mso-background-source:auto;
	mso-pattern:auto;
	mso-protection:locked visible;
	white-space:nowrap;
	mso-rotate:0;}
.xl63
	{mso-number-format:Percent;}
.xl64
	{white-space:normal;}
-->
</style>
</head>

<body link="#0563C1" vlink="#954F72">


Total No. of HHs | 276802 |  
-- | -- | --
Total   No. of persons | 744456 |  
No. of   persons who telecommute | 46692 | 6.27%
No. of   persons who belong to a HH of size >2 | 29211 | 3.92%
No. of   persons who belong to a HH which has Joint pattern | 4867 | 0.65%
No. of   persons who have at least two other people apart from telecommuters with M or   N pattern | 4867 | 0.65%
No. of   such households | 4391 | 1.59%



</body>

</html>

Reply:
<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/USVV72~1/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/USVV72~1/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>
<!--table
	{mso-displayed-decimal-separator:"\.";
	mso-displayed-thousand-separator:"\,";}
@page
	{margin:.75in .7in .75in .7in;
	mso-header-margin:.3in;
	mso-footer-margin:.3in;}
tr
	{mso-height-source:auto;}
col
	{mso-width-source:auto;}
br
	{mso-data-placement:same-cell;}
td
	{padding-top:1px;
	padding-right:1px;
	padding-left:1px;
	mso-ignore:padding;
	color:black;
	font-size:11.0pt;
	font-weight:400;
	font-style:normal;
	text-decoration:none;
	font-family:Calibri, sans-serif;
	mso-font-charset:0;
	mso-number-format:General;
	text-align:general;
	vertical-align:bottom;
	border:none;
	mso-background-source:auto;
	mso-pattern:auto;
	mso-protection:locked visible;
	white-space:nowrap;
	mso-rotate:0;}
.xl65
	{mso-number-format:Percent;}
.xl66
	{white-space:normal;}
-->
</style>
</head>

<body link="#0563C1" vlink="#954F72">


Waterfall -   Telecommute and CDAP |   |  
-- | -- | --
Total No. of HHs | 276802 |  
Total   No. of persons | 744456 |  
No. of   persons who telecommute | 46692 | 6.27%
No. of   persons who belong to a HH of size >2 | 29211 | 3.92%
No. of   persons who belong to a HH which has Joint pattern | 4867 | 0.65%
No. of   persons who have at least two other people apart from telecommuters with M or   N pattern | 4867 | 0.65%
No. of   such households | 4391 | 1.59%



</body>

</html>